runtime/ast/unmarshal.go

```diff
@@ -0,0 +1,901 @@
+	type Alias DictionaryType
+	return json.Unmarshal(b, &struct {
+		*Alias
+		KeyType   TypeUnmarshaler
+		ValueType TypeUnmarshaler
+	}{
+		Alias:     (*Alias)(a),
+		KeyType:   TypeUnmarshaler{&a.KeyType},
+		ValueType: TypeUnmarshaler{&a.ValueType},
+	})
+}
+
+func (a *DictionaryEntry) UnmarshalJSON(b []byte) error {
+	type Alias DictionaryEntry
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Key   ExpressionUnmarshaler
+		Value ExpressionUnmarshaler
+	}{
+		Alias: (*Alias)(a),
+		Key:   ExpressionUnmarshaler{&a.Key},
+		Value: ExpressionUnmarshaler{&a.Value},
+	})
+}
+
+func (a *EmitStatement) UnmarshalJSON(b []byte) error {
+	type Alias EmitStatement
+	return json.Unmarshal(b, &struct {
+		*Alias
+		StartPos *Position
+	}{
+		Alias:    (*Alias)(a),
+		StartPos: &a.StartPos,
+	})
+}
+
+func (a *ExpressionStatement) UnmarshalJSON(b []byte) error {
+	type Alias ExpressionStatement
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Expression ExpressionUnmarshaler
+	}{
+		Alias:      (*Alias)(a),
+		Expression: ExpressionUnmarshaler{&a.Expression},
+	})
+}
+
+func (a *ForStatement) UnmarshalJSON(b []byte) error {
+	type Alias ForStatement
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Value    ExpressionUnmarshaler
+		StartPos *Position
+	}{
+		Alias:    (*Alias)(a),
+		Value:    ExpressionUnmarshaler{&a.Value},
+		StartPos: &a.StartPos,
+	})
+}
+
+func (a *FunctionDeclaration) UnmarshalJSON(b []byte) error {
+	type Alias FunctionDeclaration
+	return json.Unmarshal(b, &struct {
+		*Alias
+		StartPos *Position
+	}{
+		Alias:    (*Alias)(a),
+		StartPos: &a.StartPos,
+	})
+}
+
+func (a *IfStatement) UnmarshalJSON(b []byte) error {
+	type Alias IfStatement
+	a.Test = &BoolExpression{}
+	return json.Unmarshal(b, &struct {
+		*Alias
+		StartPos *Position
+	}{
+		Alias:    (*Alias)(a),
+		StartPos: &a.StartPos,
+	})
+}
+
+func (a *ImportDeclaration) UnmarshalJSON(b []byte) error {
+	type Alias ImportDeclaration
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Location LocationUnmarshaler
+	}{
+		Alias:    (*Alias)(a),
+		Location: LocationUnmarshaler{&a.Location},
+	})
+}
+
+func (a *InstantiationType) UnmarshalJSON(b []byte) error {
+	type Alias InstantiationType
+	return json.Unmarshal(b, &struct {
+		*Alias
+		InstantiatedType TypeUnmarshaler
+		EndPos           *Position
+	}{
+		Alias:            (*Alias)(a),
+		InstantiatedType: TypeUnmarshaler{&a.Type},
+		EndPos:           &a.EndPos,
+	})
+}
+
+func (a *Members) UnmarshalJSON(b []byte) error {
+	type Alias Members
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Declarations *[]Declaration
+	}{
+		Alias:        (*Alias)(a),
+		Declarations: &a.declarations,
+	})
+}
+
+func (a *NominalType) UnmarshalJSON(b []byte) error {
+	type Alias NominalType
+	return json.Unmarshal(b, &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(a)})
+}
+
+func (a *Operation) UnmarshalJSON(b []byte) error {
+	if op, ok := operationMap[strings.Trim(string(b), `"`)]; ok {
```

Comment:
Instead of trimming the serialized string's quotes manually, potentially leaving quoted content, unmarshal as a string:

```go
var s string
if err := json.Unmarshal(b, &s); err != nil {
	return err
}
```